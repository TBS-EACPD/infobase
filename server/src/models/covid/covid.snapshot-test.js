const { execQuery } = global;

const estimates_fields = `
  est_doc
  vote
  stat
`;
const expenditures_fields = `
  month_last_updated
  vote
  stat
`;
const measure_covid_data = `
  fiscal_year

  covid_estimates {
    org_id
    org {
      id
      name
    }
  
    ${estimates_fields}
  }
  covid_expenditures {
    org_id
    org {
      id
      name
    }

    is_in_estimates
  
    ${expenditures_fields}
  }
`;
const years_with_covid_data = `
  years_with_covid_data {
    years_with_estimates
    years_with_expenditures
  }
`;

const all_covid_measures_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    covid_measures {
      id
      name

      ${years_with_covid_data}

      all_year_covid_data: covid_data {
        ${measure_covid_data}
      }

      one_year_covid_data: covid_data(fiscal_year: 2020) {
        ${measure_covid_data}
      }
    }
  }
}`;

const specific_covid_measures_query = `
query ($lang: String = "en", $covid_measure_id: String = "COV001") {
  root(lang: $lang) {
    covid_measure(covid_measure_id: $covid_measure_id) {
      id
      name

      ${years_with_covid_data}
    }
  }
}`;

const gov_covid_summary_query = `
query ($lang: String = "en", $top_x: Int = 1) {
  root(lang: $lang) {
    gov {
      all_year_summaries: covid_summary {
        fiscal_year

        top_spending_orgs {
          org_id
          name
        }
        top_spending_measures(top_x: $top_x) {
          id
          name
        }

        covid_estimates {
          ${estimates_fields}
        }
        covid_expenditures {
          ${expenditures_fields}
        }

        measure_counts {
          with_authorities
          with_spending
        }
        org_counts {
          with_authorities
          with_spending
        }
      }

      one_year_summary: covid_summary(fiscal_year: 2020) {
        fiscal_year

        top_spending_orgs {
          org_id
          name
        }
        top_spending_measures(top_x: $top_x) {
          id
          name
        }

        covid_estimates {
          ${estimates_fields}
        }
        covid_expenditures {
          ${expenditures_fields}
        }

        measure_counts {
          with_authorities
          with_spending
        }
        org_counts {
          with_authorities
          with_spending
        }
      }
    }
  }
}`;

const org_covid_summary_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    org(org_id: "133") {
      org_id
      name

      all_year_summaries: covid_summary {
        fiscal_year
        covid_estimates {
          ${estimates_fields}
        }
        covid_expenditures {
          ${expenditures_fields}
        }
      }

      one_year_summary: covid_summary(fiscal_year: 2020) {
        fiscal_year
        covid_estimates {
          ${estimates_fields}
        }
        covid_expenditures {
          ${expenditures_fields}
        }
      }
    }
  }
}`;

const org_covid_measures_query = `
query ($lang: String = "en", $org_id: String = "133") {
  root(lang: $lang) {
    org(org_id: $org_id) {
      all_year_measures: covid_measures {
        id
        name

        ${years_with_covid_data}

        all_year_covid_data: covid_data(org_id: $org_id) {
          ${measure_covid_data}
        }

        one_year_covid_data: covid_data(fiscal_year: 2020, org_id: $org_id) {
          ${measure_covid_data}
        }
      }

      one_year_measures: covid_measures(fiscal_year: 2021) {
        id
        name

        ${years_with_covid_data}

        covid_data(fiscal_year: 2021, org_id: $org_id) {
          ${measure_covid_data}
        }
      }
    }
  }
}`;

const gov_years_with_covid_data_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    gov {
      ${years_with_covid_data}
    }
  }
}`;

const org_years_with_covid_data_query = `
query ($lang: String = "en") {
  root(lang: $lang) {
    has_data: org(org_id: "133") {
      ${years_with_covid_data}
    }
    does_not_have_data: org(org_id: "15") {
      ${years_with_covid_data}
    }
    only_has_estimates: org(org_id: "1") {
      ${years_with_covid_data}
    }
  }
}`;

describe("covid data", () => {
  it("All covid measures", async () => {
    const data = await execQuery(all_covid_measures_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("Specific covid measure", async () => {
    const data = await execQuery(specific_covid_measures_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("Gov covid estimates summary", async () => {
    const data = await execQuery(gov_covid_summary_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("Org covid estimates summary", async () => {
    const data = await execQuery(org_covid_summary_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("Org covid measures", async () => {
    const data = await execQuery(org_covid_measures_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("Gov years with covid data", async () => {
    const data = await execQuery(gov_years_with_covid_data_query, {});
    return expect(data).toMatchSnapshot();
  });
  it("Org years with covid data", async () => {
    const data = await execQuery(org_years_with_covid_data_query, {});
    return expect(data).toMatchSnapshot();
  });
});
