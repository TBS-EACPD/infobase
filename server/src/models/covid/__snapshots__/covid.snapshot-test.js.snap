// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`covid data All covid estimates 1`] = `
Object {
  "data": Object {
    "root": Object {
      "covid_estimates": Array [
        Object {
          "est_doc": "ME",
          "fiscal_year": "2020",
          "id": "133-2020-ME",
          "org_id": 133,
          "stat": 8,
          "vote": 1107,
        },
        Object {
          "est_doc": "SEA",
          "fiscal_year": "2020",
          "id": "133-2020-SEA",
          "org_id": 133,
          "stat": 0,
          "vote": 9,
        },
        Object {
          "est_doc": "SEB",
          "fiscal_year": "2020",
          "id": "133-2020-SEB",
          "org_id": 133,
          "stat": 9,
          "vote": 9,
        },
        Object {
          "est_doc": "ME",
          "fiscal_year": "2020",
          "id": "326-2020-ME",
          "org_id": 326,
          "stat": 999,
          "vote": 0,
        },
        Object {
          "est_doc": "SEA",
          "fiscal_year": "2020",
          "id": "326-2020-SEA",
          "org_id": 326,
          "stat": 9,
          "vote": 99,
        },
        Object {
          "est_doc": "SEA",
          "fiscal_year": "2020",
          "id": "552-2020-SEA",
          "org_id": 552,
          "stat": 99,
          "vote": 99,
        },
        Object {
          "est_doc": "SEB",
          "fiscal_year": "2020",
          "id": "552-2020-SEB",
          "org_id": 552,
          "stat": 9,
          "vote": 18,
        },
      ],
    },
  },
}
`;

exports[`covid data All covid initiatives and initiative estimates 1`] = `
Object {
  "data": Object {
    "root": Object {
      "covid_initiatives": Array [
        Object {
          "covid_initiative_estimates": Array [
            Object {
              "covid_measure_ids": Array [
                "COV001",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
              ],
              "est_doc": "ME",
              "fiscal_year": "2020",
              "id": "133-2020-ME-1",
              "org_id": 133,
              "stat": 0,
              "vote": 999,
            },
            Object {
              "covid_measure_ids": Array [
                "COV001",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
              ],
              "est_doc": "SEA",
              "fiscal_year": "2020",
              "id": "133-2020-SEA-1",
              "org_id": 133,
              "stat": 0,
              "vote": 9,
            },
            Object {
              "covid_measure_ids": Array [
                "COV001",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
              ],
              "est_doc": "SEB",
              "fiscal_year": "2020",
              "id": "133-2020-SEB-1",
              "org_id": 133,
              "stat": 9,
              "vote": 9,
            },
            Object {
              "covid_measure_ids": Array [
                "COV001",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
              ],
              "est_doc": "ME",
              "fiscal_year": "2020",
              "id": "326-2020-ME-1",
              "org_id": 326,
              "stat": 999,
              "vote": 0,
            },
            Object {
              "covid_measure_ids": Array [
                "COV001",
                "COV002",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
                Object {
                  "id": "COV002",
                  "name": "measure 2",
                },
              ],
              "est_doc": "SEA",
              "fiscal_year": "2020",
              "id": "552-2020-SEA-1",
              "org_id": 552,
              "stat": 99,
              "vote": 99,
            },
            Object {
              "covid_measure_ids": Array [
                "COV001",
                "COV002",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
                Object {
                  "id": "COV002",
                  "name": "measure 2",
                },
              ],
              "est_doc": "SEB",
              "fiscal_year": "2020",
              "id": "552-2020-SEB-1",
              "org_id": 552,
              "stat": 0,
              "vote": 9,
            },
          ],
          "id": "1",
          "name": "Initiative 1",
        },
        Object {
          "covid_initiative_estimates": Array [
            Object {
              "covid_measure_ids": Array [
                "COV002",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV002",
                  "name": "measure 2",
                },
              ],
              "est_doc": "ME",
              "fiscal_year": "2020",
              "id": "133-2020-ME-2",
              "org_id": 133,
              "stat": 8,
              "vote": 99,
            },
            Object {
              "covid_measure_ids": Array [
                "COV002",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV002",
                  "name": "measure 2",
                },
              ],
              "est_doc": "SEB",
              "fiscal_year": "2020",
              "id": "552-2020-SEB-2",
              "org_id": 552,
              "stat": 9,
              "vote": 9,
            },
          ],
          "id": "2",
          "name": "Initiative 2",
        },
        Object {
          "covid_initiative_estimates": Array [
            Object {
              "covid_measure_ids": Array [
                "COV001",
                "COV002",
                "COV003",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
                Object {
                  "id": "COV002",
                  "name": "measure 2",
                },
                Object {
                  "id": "COV003",
                  "name": "measure 3",
                },
              ],
              "est_doc": "ME",
              "fiscal_year": "2020",
              "id": "133-2020-ME-3",
              "org_id": 133,
              "stat": 0,
              "vote": 9,
            },
            Object {
              "covid_measure_ids": Array [
                "COV001",
                "COV002",
                "COV003",
              ],
              "covid_measures": Array [
                Object {
                  "id": "COV001",
                  "name": "measure 1",
                },
                Object {
                  "id": "COV002",
                  "name": "measure 2",
                },
                Object {
                  "id": "COV003",
                  "name": "measure 3",
                },
              ],
              "est_doc": "SEA",
              "fiscal_year": "2020",
              "id": "326-2020-SEA-3",
              "org_id": 326,
              "stat": 9,
              "vote": 99,
            },
          ],
          "id": "3",
          "name": "Initiative 3",
        },
      ],
    },
  },
}
`;

exports[`covid data All covid measures 1`] = `
Object {
  "data": Object {
    "root": Object {
      "covid_measures": Array [
        Object {
          "id": "COV001",
          "name": "measure 1",
        },
        Object {
          "id": "COV002",
          "name": "measure 2",
        },
        Object {
          "id": "COV003",
          "name": "measure 3",
        },
      ],
    },
  },
}
`;

exports[`covid data Org covid estimates 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "covid_estimates": Array [
          Object {
            "est_doc": "ME",
            "fiscal_year": "2020",
            "id": "133-2020-ME",
            "stat": 8,
            "vote": 1107,
          },
          Object {
            "est_doc": "SEA",
            "fiscal_year": "2020",
            "id": "133-2020-SEA",
            "stat": 0,
            "vote": 9,
          },
          Object {
            "est_doc": "SEB",
            "fiscal_year": "2020",
            "id": "133-2020-SEB",
            "stat": 9,
            "vote": 9,
          },
        ],
      },
    },
  },
}
`;

exports[`covid data Org covid initiatives and initiative estimates 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "covid_initiatives": Array [
          Object {
            "covid_initiative_estimates": Array [
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                ],
                "est_doc": "ME",
                "fiscal_year": "2020",
                "id": "133-2020-ME-1",
                "org_id": 133,
                "stat": 0,
                "vote": 999,
              },
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                ],
                "est_doc": "SEA",
                "fiscal_year": "2020",
                "id": "133-2020-SEA-1",
                "org_id": 133,
                "stat": 0,
                "vote": 9,
              },
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                ],
                "est_doc": "SEB",
                "fiscal_year": "2020",
                "id": "133-2020-SEB-1",
                "org_id": 133,
                "stat": 9,
                "vote": 9,
              },
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                ],
                "est_doc": "ME",
                "fiscal_year": "2020",
                "id": "326-2020-ME-1",
                "org_id": 326,
                "stat": 999,
                "vote": 0,
              },
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                  "COV002",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                  Object {
                    "id": "COV002",
                    "name": "measure 2",
                  },
                ],
                "est_doc": "SEA",
                "fiscal_year": "2020",
                "id": "552-2020-SEA-1",
                "org_id": 552,
                "stat": 99,
                "vote": 99,
              },
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                  "COV002",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                  Object {
                    "id": "COV002",
                    "name": "measure 2",
                  },
                ],
                "est_doc": "SEB",
                "fiscal_year": "2020",
                "id": "552-2020-SEB-1",
                "org_id": 552,
                "stat": 0,
                "vote": 9,
              },
            ],
            "id": "1",
            "name": "Initiative 1",
          },
          Object {
            "covid_initiative_estimates": Array [
              Object {
                "covid_measure_ids": Array [
                  "COV002",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV002",
                    "name": "measure 2",
                  },
                ],
                "est_doc": "ME",
                "fiscal_year": "2020",
                "id": "133-2020-ME-2",
                "org_id": 133,
                "stat": 8,
                "vote": 99,
              },
              Object {
                "covid_measure_ids": Array [
                  "COV002",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV002",
                    "name": "measure 2",
                  },
                ],
                "est_doc": "SEB",
                "fiscal_year": "2020",
                "id": "552-2020-SEB-2",
                "org_id": 552,
                "stat": 9,
                "vote": 9,
              },
            ],
            "id": "2",
            "name": "Initiative 2",
          },
          Object {
            "covid_initiative_estimates": Array [
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                  "COV002",
                  "COV003",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                  Object {
                    "id": "COV002",
                    "name": "measure 2",
                  },
                  Object {
                    "id": "COV003",
                    "name": "measure 3",
                  },
                ],
                "est_doc": "ME",
                "fiscal_year": "2020",
                "id": "133-2020-ME-3",
                "org_id": 133,
                "stat": 0,
                "vote": 9,
              },
              Object {
                "covid_measure_ids": Array [
                  "COV001",
                  "COV002",
                  "COV003",
                ],
                "covid_measures": Array [
                  Object {
                    "id": "COV001",
                    "name": "measure 1",
                  },
                  Object {
                    "id": "COV002",
                    "name": "measure 2",
                  },
                  Object {
                    "id": "COV003",
                    "name": "measure 3",
                  },
                ],
                "est_doc": "SEA",
                "fiscal_year": "2020",
                "id": "326-2020-SEA-3",
                "org_id": 326,
                "stat": 9,
                "vote": 99,
              },
            ],
            "id": "3",
            "name": "Initiative 3",
          },
        ],
      },
    },
  },
}
`;
